crates/dht/src/hash.rs

```diff
@@ -0,0 +1,443 @@
+//! Partition of the hash space.
+//!
+//! The space of possible hashes is mapped to a 32-bit location. See [::kitsune2_api::id::Id::loc] for
+//! more information about this. Each agent is responsible for storing and serving some part of
+//! the hash space. This module provides a structure that partitions the hash space by some factor.
+//!
+//! The factor must be chosen up front and cannot be changed. The factor determines the number of
+//! partitions to create. A factor of 32 will create a single partition. Each unit decrease from 32
+//! will double the number of partitions. A good default for a network that is expected to grow to
+//! a few thousand nodes is 21. This will create 2^11 partitions, which is 2048. Given that data is
+//! also stored redundantly, say 10 times, this means that 20,000 nodes are needed for the data
+//! to be maximally distributed. With higher replication, say 50 times, and a hash factor of 20,
+//! over 200,000 nodes are needed for maximal distribution.
+//!
+//! Each hash partition manages a [PartitionedTime] structure that is responsible for managing the
+//! time slices for that hash partition. The interface of this module is largely responsible for
+//! delegating the updating of time slices to the inner time partitions. This ensures that all the
+//! time partitions are updated in lockstep. This makes reasoning about the space-time partitioning
+//! easier.
```

Comment:
WDYT about using "space partition" for arcs and "time slice" and "time partition" for the time divisions?
```suggestion
//! Each space partition manages a [PartitionedTime] structure that is responsible for managing the
//! time slices for that space partition. The interface of this module is largely responsible for
//! delegating the updating of time slices to the inner time partitions. This ensures that all the
//! time partitions are updated in lockstep. This makes reasoning about the space-time partitioning
//! easier.
```